3-UsingAPIs/Week2/assignment/ex3-rollAnAce.js

```diff
@@ -17,24 +17,26 @@ import { rollDie } from '../../helpers/pokerDiceRoller.js';
  * @param {DieFace} desiredValue
  * @returns {Promise<DieFace>}
  */
-export function rollDieUntil(desiredValue) {
-  // TODO rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== desiredValue) {
-      return rollDieUntil(desiredValue);
+export async function rollDieUntil(desiredValue) {
+  while (true) {
+    const value = await rollDie();
+    if (value === desiredValue) {
+      return value;
     }
-    return value;
-  });
+  }
 }
```

Comment:
There is no need to create an endless loop. You can use the loop condition instead of a separate `if` statement and simplify the code as follows:

```suggestion
export async function rollDieUntil(desiredValue) {
  let value;
  while (value !== desiredValue) {
    value = await rollDie();
  }
  return value;
}
```